torch/_inductor/fx_passes/pad_mm.py

```diff
@@ -205,35 +256,61 @@ def tensor_key(t):
 def should_pad_bench(
     mat1: Tensor, mat2: Tensor, op, input: Optional[Tensor] = None
 ) -> bool:
-    if not has_triton():
-        return False
+    if torch._inductor.config.shape_pad_always:
+        return True
 
     do_bench = functools.partial(
         utils.do_bench,
         warmup=5,
     )
-
+    m_padded_length = 0
+    n_padded_length = 0
+    batchsize = 1
+    explicit_transpose = False
     with no_dispatch():
         if op is torch.ops.aten.mm or op is torch.ops.aten.addmm:
             m = mat1.shape[0]
             k = mat1.shape[1]
             n = mat2.shape[1]
-
-            m_padded_length = get_padded_length(m, get_alignment_size(mat1))
             k_padded_length = get_padded_length(k, get_alignment_size(mat1))
-            n_padded_length = get_padded_length(n, get_alignment_size(mat2))
+            if not torch._inductor.config.shape_pad_only_k_dim:
+                n_padded_length = get_padded_length(n, get_alignment_size(mat2))
+                m_padded_length = get_padded_length(m, get_alignment_size(mat1))
         elif op is torch.ops.aten.bmm:
+            batchsize = mat1.shape[0]
             m = mat1.shape[1]
-            k = mat2.shape[2]
+            k = mat1.shape[2]
```

Comment:
this was just fixed by @shunting314. undo change here ?

Reply:
I guess you're referring to this: https://github.com/pytorch/pytorch/pull/118598/commits/064fb7d1ac1c2b1487c4e6225c09b0c39f0db153 - So, this diff's version of the line and @shunting314 version are identical. I didn't even notice it was a preexisting bug, just thought I made a typo and fixed it while I was at it ;)